### PR TITLE
Add config for disabling use of root network namespace

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?
Adds a new config `system_probe_config.enable_root_netns` to enable/disable use of the root network namespace.

### Describe how to test/QA your changes
QA instructions here for testing against a test in the code base, so you will have to sync to the release branch being tested on a linux VM.
1. From the top of the repo, run: ```go test -tags linux_bpf ./pkg/network/config -c```; this will create a test binary `config.test` 
2. create a new network namespace: ```sudo ip netns add test```
3. run the test in the new namespace: ```sudo ip netns exec test ./config.test -test.run TestDisableRootNetNamespace -test.v -test.count 1```
4. cleanup network namespace: ```sudo ip netns del test```

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/system_probe.go [hunk 1]
```diff
@@ -165,6 +165,9 @@ func InitSystemProbeConfig(cfg Config) {
 
 	// service monitoring
 	cfg.BindEnvAndSetDefault(join(smNS, "enabled"), false, "DD_SYSTEM_PROBE_SERVICE_MONITORING_ENABLED")
+
+	// enable/disable use of root net namespace
+	cfg.BindEnvAndSetDefault(join(netNS, "disable_root_netns"), false)
```
Comment (brycekahle) [inline pkg/config/system_probe.go:170]: Can we flip it to an enable flag, with the default being true? I find it easier to read that way.

File: pkg/network/config/config.go [hunk 2]
```diff
@@ -221,6 +237,8 @@ func New() *Config {
 		EnableMonotonicCount: cfg.GetBool(join(spNS, "windows.enable_monotonic_count")),
 
 		RecordedQueryTypes: cfg.GetStringSlice(join(netNS, "dns_recorded_query_types")),
+
+		DisableRootNetNs: cfg.GetBool(join(spNS, "disable_root_netns")),
```
Comment (brycekahle) [inline pkg/network/config/config.go:241]:
```suggestion
		DisableRootNetNs: cfg.GetBool(join(netNS, "disable_root_netns")),
```

File: pkg/config/system_probe.go (hunk 1 above)
Comment (brycekahle) [reply pkg/config/system_probe.go:170]: Maybe name it `use_root_netns`?

**Review by brycekahle — APPROVED** [review]
This looks good. Any ideas on how we could test this to ensure we don't break the option in the future?

**Comment by hmahmood** [comment]
> This looks good. Any ideas on how we could test this to ensure we don't break the option in the future?

I added a very basic test, but without being able to run the test in a different process or network namespace, we can't add an automated test. The test I added can be run manually in a different namespace, so I'll add instructions for that for qa.

**Review by brycekahle — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

**Pull request merged**